### PR TITLE
Add ckb-vm fuzz targets

### PR DESCRIPTION
**Pull request by mohanson**
- Fuzz ckb-vm with spike-sys
- Provides an OOP spike-sys interface

### CODE REVIEW

File: cpp/spike-interfaces.cc [hunk 1]
```diff
@@ -57,6 +57,7 @@ uint64_t rvv_new_processor(uint32_t vlen, uint32_t elen, uint64_t mem_size) {
     processor_t *proc = new processor_t("RV64GCV", "MSU", buf, mem, 0, false, NULL, std::cerr);
     reg_t val = proc->state.sstatus->read();
     proc->state.sstatus->write(val | SSTATUS_VS);
+    proc->VU.vxrm->write(0x02);
```
Comment (XuJiandong) [inline cpp/spike-interfaces.cc:60]: What is this used for?

File: fuzz/fuzz_targets/fuzz_alu.rs [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+        [0b000000_1_11111_11111_000_11111_0000000, 0b001100_0_00000_00000_100_00000_1010111], // vrgather.vx
+        [0b000000_1_11111_11111_000_11111_0000000, 0b001100_0_00000_00000_011_00000_1010111], // vrgather.vi
+        [0b000000_0_11111_11111_000_11111_0000000, 0b010111_1_00000_00000_010_00000_1010111], // vcompress.vm
+        [0b000000_0_11111_00000_000_11111_0000000, 0b100111_1_00000_00000_011_00000_1010111], // vmv1r.v
+        [0b000000_0_11111_00000_000_11111_0000000, 0b100111_1_00000_00001_011_00000_1010111], // vmv2r.v
+        [0b000000_0_11111_00000_000_11111_0000000, 0b100111_1_00000_00011_011_00000_1010111], // vmv4r.v
+        [0b000000_0_11111_00000_000_11111_0000000, 0b100111_1_00000_00111_011_00000_1010111], // vmv8r.v
+    ];
+
+    for _ in 0..128 {
```
Comment (XuJiandong) [inline fuzz/fuzz_targets/fuzz_alu.rs:379]: With so many loops, How can we find the root cause if there is any error?

File: fuzz/fuzz_targets/fuzz_mem.rs [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+                    ckbvm.element_mut(2, 16, i as usize).copy_from_slice(buf);
+                }
+                0b110 => {
+                    let offset = [0u64, 8][rand.u64() as usize % 2];
+                    spike.set_vreg(32 + i * 4, (&offset) as *const u64 as *const u8, 4).unwrap();
+                    let buf = unsafe { std::slice::from_raw_parts((&offset) as *const u64 as *const u8, 4) };
+                    ckbvm.element_mut(2, 32, i as usize).copy_from_slice(buf);
+                }
+                0b111 => {
+                    let offset = [0u64, 8][rand.u64() as usize % 2];
```
Comment (XuJiandong) [inline fuzz/fuzz_targets/fuzz_mem.rs]: I think, the randomness of offset is not enough.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cpp/spike-interfaces.cc (hunk 1 above)
Comment (mohanson) [reply cpp/spike-interfaces.cc:60]: Set rounding mode to `round-down (truncate)`

File: fuzz/fuzz_targets/fuzz_alu.rs (hunk 2 above)
Comment (mohanson) [reply fuzz/fuzz_targets/fuzz_alu.rs:379]: Rerun artifacts/xxx with environment `LOG=1`

File: fuzz/fuzz_targets/fuzz_mem.rs (hunk 3 above)
Comment (mohanson) [reply fuzz/fuzz_targets/fuzz_mem.rs]: yes, it can be optimized

Comment (mohanson) [reply fuzz/fuzz_targets/fuzz_mem.rs]: https://github.com/riscv-software-src/riscv-isa-sim/blob/master/riscv/mmu.h#L210-L216

There is a trade-off here, spike requires aligned addresses by default, so I can't make offset equals to 1, 2, 3, 4, 5, 6, 7 ...

Comment (XuJiandong) [reply fuzz/fuzz_targets/fuzz_mem.rs]: Can we have some other offset values? Like 16, 24?

Comment (mohanson) [reply fuzz/fuzz_targets/fuzz_mem.rs]: Updated.

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**